docs/proposals/scheduling/multi-podtemplate-support/multiple-pod-template-support.md

```diff
@@ -0,0 +1,195 @@
+
+## Background on our Use-Case
+
+Karmada will be used as an intelligent scheduler for FlinkDeployments. We aim to use the accurate estimator (with the
+ResourceQuota plugin enabled), to estimate whether a FlinkDeployment can be fully scheduled on the potential destination namespace.
+In order to make this estimation, we need to take into account all of the resource requirements of the components that will be
+scheduled by the Flink Operator. Once the CRD is scheduled by Karmada, the Flink Operator will take over the rest of the component
+scheduling as seen below.
+
+![Karmada-Scheduler](Karmada-Scheduler.png)
+
+In the case of Flink, these components are the JobManager(s) as well as the TaskManager(s). Both of these components can be comprised of
+multiple pods, and the JM and TM frequently do not have the same resource requirements.
+
+## Motivation
+
+Karmada currently provides 2 methods of scheduling estimation through:
+1. The general estimator (which analyzes total cluster resources to determine scheduling)
+2. The accurate estimator (which can inspect namespaced resource quotas and determine
+   number of potential replicas via the ResourceQuota plugin)
+
+This proposal aims to improve the 2nd method by allowing users to define components for their replica
+and provide precise resourceRequirements.
+
+## Goals
+
+- Provide a declarative pattern for defining the resourceRequests for individual replica components
+- Allow more accurate scheduling estimates for CRDs
+
+## Design Details
+
+### API change
+
+The main changes of this proposal are to the API definition of the ReplicaRequirements struct. We currently include the replicaCount and
+replicaRequirements as root level attributes to the ResourceBindingSpec. The limitation here is that we are unable to define unique
+replicaRequirements in the case that the resource has more than one podTemplate.
+
+To address this, we can move the concept of replicas and replicaRequirements into a struct related to the individual resource's `Components`.
+
+Each `Component` will have a `Name`, the number of `Replicas`, and corresponding `replicaRequirements`.
+These basic fields are necessary to allow the accurate estimator to determine whether all components of the CRD replica
+will be able to fit on the destination namespace.
+
+The definition of ReplicaRequirements will stay the same - with the drawback that the user will need to define how Karmada
+interprets the individual components of the CRD. Karmada should also support a default component which will use one of the resource's
+podTemplates to find requirements.
+
+```go
+
+type ResourceBindingSpec struct {
+
+    . . .
+
+    // The total number of replicas scheduled by this resource. Each replica will represented by exactly one component of the resource.
+    TotalReplicas int32 `json:"totalReplicas,omitempty"`
```

Comment:
I've included this field as a replacement for the existing `Replicas` field, which is used very frequently within the Karmada codebase. Even though we are introducing the concept of components, Karmada will still ultimately be scheduling replicas - so I believe this slight refactor will make the implementation of this change less complex. This is again making an assumption that resources with more than 1 component will be scheduled on the same cluster.